### PR TITLE
[onert] Implement forward and backward functions in TrainableFnSequence

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit implements forward and backward functions in TrainableFnSequence.
Train backend kernel will append trainable functions using this append api.

ONE-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

waiting #10972

### CODE REVIEW

**Comment by jyoungyun** [comment]
@nnfw-bot test onert-x64-release

**Comment by jyoungyun** [comment]
@nnfw-bot test onert-x64-release

File: runtime/onert/core/src/exec/train/TrainableFnSequence.cc [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+void TrainableFnSequence::backward()
+{
+  for (auto it = _functions.rbegin(); it != _functions.rend(); ++it)
+  {
+    (*it)->backward();
+  }
+}
+
+void TrainableFnSequence::append(std::unique_ptr<ITrainableFunction> &&function)
```
Comment (ragmani) [inline runtime/onert/core/src/exec/train/TrainableFnSequence.cc:42]: (optional)
```suggestion
void TrainableFnSequence::append(std::unique_ptr<ITrainableFunction> function)
```

File: runtime/onert/core/src/exec/train/TrainableFnSequence.cc [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+{
+  _functions.push_back(std::move(function));
+}
+
+void TrainableFnSequence::iterate(const std::function<void(ITrainableFunction &)> &fn)
```
Comment (ragmani) [inline runtime/onert/core/src/exec/train/TrainableFnSequence.cc:47]: I didn't understand the purpose of this method. Is it necessary?

**Review by ragmani — APPROVED** [review]
LGTM

File: runtime/onert/core/src/exec/train/TrainableFnSequence.cc (hunk 2 above)
Comment (jyoungyun) [reply runtime/onert/core/src/exec/train/TrainableFnSequence.cc:47]: This code is related to [this code](https://github.com/Samsung/ONE/blob/ed69577f8282128dc64129ae3c17bf4aaf20c23d/runtime/onert/backend/train/BackendContext.cc#L91).
I'm not sure what method to be called when creating kernel. However, this function is required to enable the code.

File: runtime/onert/core/src/exec/train/TrainableFnSequence.cc (hunk 1 above)
Comment (jyoungyun) [reply runtime/onert/core/src/exec/train/TrainableFnSequence.cc:42]: (out of curiosity)
Why do you suggest removing the `&&` operator?

**Review by hseok-oh — APPROVED** [review]
LGTM

**Pull request merged**